src/type-api.ts

```diff
@@ -0,0 +1,28 @@
+import FolioAPI from "./folio-api.js"
+
+export default class TypeAPI extends FolioAPI {
+  async getMapFor<Type>(path: string, { key = undefined, cache }: Partial<{ key: string, cache: Map<string, Map<string, any>> }>): Promise<Map<string, Type>> {
```

Comment:
some type aliases in this file might make this signature and the one on line 19 a little more readable

Reply:
Happy to pair up on this, or if you just want to toss a commit on the PR 🤷‍♂️